apluslms-yamlidator/apluslms_yamlidator/document.py

```diff
@@ -38,6 +40,16 @@ def find_ml(current, keys):
             current = current.setdefault(key, {})
     return current, keys[-1]
 
+def update(data, new_data):
+    for key in new_data:
+        if key in data:
+            if isinstance(data[key], list) and isinstance(new_data[key], list):
+                data[key].extend(new_data[key])
+                continue
+            if isinstance(data[key], dict) and isinstance(new_data[key], dict):
```

Comment:
Should test that `isinstance(plaah, (dict, MutableMapping)`, new_data of course could be just `Mapping`.

Reply:
so `isinstance(new_data[key], (dict, MutableMapping, Mapping))`, or what exactly did you mean?

Reply:
MutableMapping is always a subclass of Mapping. One of them is fine. `new_data` could be just a Mapping as we don't need to write to it, though I don't know any existing instances of Mapping, only MutableMappings. `data` needs to be mutable for us to be able to write to it.